test/unit/spec/widgets/kbaseTabs-spec.js

```diff
@@ -8,13 +11,283 @@ define(['jquery', 'kbaseTabs'], ($, KBaseTabs) => {
+            expect($host.text()).toContain('BAR');
+            expect($host.text()).toContain('Baz');
+            expect($host.text()).not.toContain('Fuzz');
+        });
+
+        async function expectSelectNthTab($host, n, content) {
+            const $tab = $($host.find(`[role="tab"]:nth-child(${n})`));
+            expect($tab.text()).toContain(content.nav);
+            $tab.find('a').click();
+            return await tryFor(() => {
+                const foundIt = $host.text().includes(content.panel);
+                return [foundIt, null];
+            }, 1000);
+        }
+
+        async function expectCloseNthTab($host, n, content) {
+            const $tab = $($host.find(`[role="tab"]:nth-child(${n})`));
+            expect($tab.text()).toContain(content.closedTab);
+            $tab.find('button').click();
+
+            // We only check for success if content was passed in, otherwise
+            // we are just clicking it.
+            if (content.nextTab) {
+                return await tryFor(() => {
+                    const fooGone = !$host.text().includes(content.closedTab);
+                    const fuzzHere = $host.text().includes(content.nextTab);
+                    return [fooGone && fuzzHere, null];
+                }, 1000);
+            }
+        }
+
+        async function expectCloseNthTabWithDialog($host, n, content) {
+            const $tab = $($host.find(`[role="tab"]:nth-child(${n})`));
+            expect($tab.text()).toContain(content.closedTab);
+            $tab.find('button').click();
+
+            // First we look for the modal and confirm the deletion.
+            await tryFor(() => {
+                const $deleteButton = $('.modal-dialog .btn:contains(Delete)');
+                if ($deleteButton.length === 0) {
+                    return [false, false];
+                }
+                $deleteButton.click();
+                return [true, true];
+            }, 1000);
+
+            return await tryFor(() => {
+                const fooGone = !$host.text().includes(content.closedTab);
+                const fuzzHere = $host.text().includes(content.nextTab);
+                return [fooGone && fuzzHere, null];
+            }, 1000);
+        }
+
+        it('should render a tab pane when the tab button is clicked', async () => {
+            const $host = $(document.createElement('div'));
+            new KBaseTabs($host, {
+                tabs: [
+                    {
+                        tab: 'Foo',
+                        content: 'BAR',
+                    },
+                    {
+                        tab: 'Baz',
+                        content: 'Fuzz',
+                    },
+                ],
+            });
+            expect($host.text()).toContain('Foo');
+            expect($host.text()).toContain('BAR');
+            expect($host.text()).toContain('Baz');
+            expect($host.text()).not.toContain('Fuzz');
```

Comment:
since this is repeated several times, can you extract it out into a function?

Reply:
If you look closely you'll see that each test case configures the tabs differently.